src/ias_proxy/src/ias_proxy_server/mod.rs

```diff
@@ -0,0 +1,209 @@
+ limitations under the License.
+------------------------------------------------------------------------------
+*/
+
+extern crate futures;
+extern crate hyper;
+extern crate ias_client;
+extern crate serde;
+extern crate serde_json;
+extern crate tokio_core;
+
+use lru_cache::LruCache;
+use self::futures::Future;
+use self::futures::future;
+use self::hyper::{Body, Error, Method, Request, Response, Server, StatusCode};
+use self::hyper::header::{HeaderMap, HeaderName, HeaderValue};
+use self::hyper::service::service_fn;
+/// module ias_proxy_server, has code to start proxy server
+use self::ias_client::IasClient;
+use self::tokio_core::reactor::Core;
+use std::collections::HashMap;
+use std::net::SocketAddr;
+use std::str::FromStr;
+use std::sync::Mutex;
+
+type ResponseBox = Box<Future<Item=Response<Body>, Error=Error> + Send>;
+
+/// structure defining IAS proxy server
+pub struct IasProxyServer {
+    ias_proxy_name: String,
+    ias_proxy_port: String,
+    ias_client: IasClient,
+}
+
+/// response body from the IAS
+#[derive(Deserialize)]
+struct IasResponseBody {
+    #[serde(rename = "isvEnclaveQuote")]
+    isv_enclave_quote: String,
+    #[serde(rename = "pseManifest")]
+    pse_manifest: String,
+    nonce: u64,
+}
+
+lazy_static! {
+    static ref sig_rl_cache: Mutex<LruCache> = Mutex::new(LruCache::new(None));
+    static ref attestation_cache: Mutex<LruCache> = Mutex::new(LruCache::new(None));
+}
+
+impl IasProxyServer {
+    /// Create new instance of IasProxyServer
+    pub fn new(config_map: HashMap<String, String>) -> Self {
+        IasProxyServer {
+            ias_proxy_name: config_map["proxy_name"].clone(),
+            ias_proxy_port: config_map["proxy_port"].clone(),
+            ias_client: IasClient::new(config_map["ias_url"].clone(), Vec::from(config_map["spid_cert_file"].as_bytes()), None),
+        }
+    }
+
+    /// run method to start listeing on the identified port and IP
+    pub fn run(&self) -> () {
+        // Start the web server on the configured URL
+        let path = self.ias_proxy_name.as_str().to_owned() + ":" + &self.ias_proxy_port.as_str().to_owned();
+        info!("Path constructed is {}", path);
+        let socket_addr: SocketAddr = match SocketAddr::from_str(&path) {
+            Ok(found_addr) => found_addr,
+            Err(err) => panic!("Error reading the address: {}", err),
+        };
+        let ias_client = self.ias_client.clone();
+        // TODO: Store this server instance and call shutdown
+        // TODO: Solve below error (logic error) using mutex
+        let new_service = move || {
+            let ias_client = ias_client.clone();
+            service_fn(move |req|
+                respond_to_request(req, /*&sig_rl_cache, &attestation_cache, */&ias_client)
+            )
+        };
+        let server = Server::bind(&socket_addr).serve(new_service);
+        hyper::rt::run(server.map_err(|e| {
+            panic!("Server error: {}", e);
```

Comment:
Is **panic** really pertinent whenever used? Can logging an error message suffice?

Reply:
I didn't know idiomatic way of error handling in Rust. Please suggest a reference link, will also talk to you on this.